src/redux/books/books.js

```diff
@@ -50,13 +51,12 @@ const booksSlice = createSlice({
             category: action.payload[id][0].category,
             progress: 0,
           };
-          state.push(book);
+          bookList.push(book);
         });
+        return bookList;
       })
-      .addCase(deleteBook.fulfilled, (state, action) => {
-        const index = state.find((book) => book.id === action.payload.id);
-        state.splice(index, 1);
-      });
+      // eslint-disable-next-line
```

Comment:
Kindly avoid disabling linter checks inline because linters represent coding style that you share with your team. They most of the time also enforce best coding practices.